Extensions.cs

```diff
@@ -1,26 +1,26 @@
-﻿using System;
-using System.Collections.Generic;
+﻿using System.Collections;
 
 namespace Boot.Codes
 {
     public static class Extensions
     {
-        private static readonly Random WeakRng = new Random();
-
         /**
          * Shuffle the array in-place.
          */
-        public static void Shuffle<T>(this IList<T> list)
+        public static T Shuffle<T>(this T list) where T : IList
         {
             // Implementation of the Fisher-Yates shuffle
             var size = list.Count;
             for (var i = 0; i < size - 2; i++)
             {
-                var j = WeakRng.Next(i, size);
+                var j = StrongRandom.Next(i, size);
                 var value = list[i];
                 list[i] = list[j];
                 list[j] = value;
             }
+
+            // allows method chaining
+            return list;
```

Comment:
I still disagree that this method should have a return value. You're passing the input by reference, modifying it in place, then *also* return the input argument again. This is a footgun for the next person who uses this code